tkg/test/tkgctl/shared/utils.go

```diff
@@ -0,0 +1,71 @@
+package shared
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	admissionregistrationv1 "k8s.io/client-go/kubernetes/typed/admissionregistration/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	webhookScrtName   = "webhook-tls"
+	systemNamespace   = "tkg-system"
+	webhookLabelKey   = "tkg.tanzu.vmware.com/addon-webhooks"
+	webhookLabelValue = "addon-webhooks"
+)
+
+// getWebhookCACert returns the value of CA certificate "ca-cert.pem" for the "webhook-tls" secret
+func getWebhookCACert(ctx context.Context, k8sClient client.Client, secretName, secretNamespace string) (string, error) {
+	secret := corev1.Secret{}
+	if err := k8sClient.Get(ctx, types.NamespacedName{Name: secretName, Namespace: secretNamespace}, &secret); err != nil {
+		return "", err
+	}
+	if value, exists := secret.Data["ca-cert.pem"]; exists {
+		return string(value), nil
+	}
+	return "", nil
```

Comment:
Is the scenario where the `ca-cert.pem` entry does not exist in the secret expected?

Reply:
Oh no:-/ I should either return an error here, or do ```Expect(caCert).NotTo(BeEmpty())``` on the caller. Thanks for catching it